src/toolchain/toolchain.ts

```diff
@@ -156,4 +162,33 @@ export class SwiftToolchain {
             throw Error("Cannot find swift executable.");
         }
     }
+
+    /**
+     * @returns if the default Swift driver is the new driver
+     */
+    private static async judgeNewDriver(toolchainPath: string): Promise<boolean> {
+        const toolDirectory = path.join(toolchainPath, "usr", "bin");
+        // judge from environment variable
+        if (process.env.SWIFT_USE_NEW_DRIVER) {
+            return true;
+        }
+        if (process.env.SWIFT_USE_OLD_DRIVER) {
+            return false;
+        }
+        // judge from tool existence
+        if (await pathExists(toolDirectory, getExecutableName("swift-driver"))) {
+            return true;
+        }
+        if (await !pathExists(toolDirectory, getExecutableName("swift-frontend"))) {
```

Comment:
Is there any situation where `swift-driver` doesn't exist and the new driver is used.

Reply:
Yes there is — when `swift` is replaced by the new driver.